### ISSUE TITLE
Remove v1beta2 API

### ISSUE THREAD

**Issue by jbartosik** [kubernetes/autoscaler#5922]
Follow up to #4969, remove the deprecated API.

> Beta API versions must be supported for 9 months or 3 releases (whichever is longer) after deprecation

Deprecation was in [VPA 0.13.0](https://github.com/kubernetes/autoscaler/releases/tag/vertical-pod-autoscaler-0.13.0), which was released on 2023-02-03; In the mean time we had [VPA 0.14.0](https://github.com/kubernetes/autoscaler/releases/tag/vertical-pod-autoscaler-0.14.0).

So  the next VPA can't drop the API yet (it will be 3rd one). We also need to wait at least until 2023-11-03 before dropping the API.

So likely the release triggered by K8s 1.28 in August keeps the API.

And the release triggered by K8s 1.29 can drop it.

**Comment by Shubham82** [comment]
Any update on this issue, as current k8s version is k8s v1.31and v1beta2 API (with depricated message) is still present in [vpa-v1-crd-gen.yaml](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/deploy/vpa-v1-crd-gen.yaml) file.

**Comment by Shubham82** [comment]
cc @jbartosik @voelzmo @kwiesmueller

**Comment by voelzmo** [comment]
Agreed, we should be able to drop this now and document that the new release won't work with k8s versions < v1.29

**Comment by raywainman** [comment]
SGTM

**Comment by adrianmoisey** [comment]
> Agreed, we should be able to drop this now and document that the new release won't work with k8s versions < v1.29

Isn't the removal of the CRD/API related to the VPA version and not the Kubernetes version?

**Comment by raywainman** [comment]
We've been trying to release a VPA version roughly every k8s minor version so that's likely what Joachim was meaning here.

**Comment by adrianmoisey** [comment]
> Since that's not a promise we make to users (directly or indirectly)

I see we do have it written down here: https://github.com/kubernetes/autoscaler/blob/16210e3da03def7e7cd5cc55012dd2ff6ffd8683/vertical-pod-autoscaler/RELEASE.md#L3-L4

So we do kinda promise it to the user, I guess.
